<a href="https://colab.research.google.com/github/Dinadedt/2022/blob/main/CICLI%231_MATEMATICOSpython.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
print("Hola Mundo", "Bienvenidos a Programacion Computacional III")


Hola Mundo Bienvenidos a Programacion Computacional III


In [ ]:
print("Luis Hernandez")

Luis Hernandez


In [ ]:
nombre = "Luis Hernandez"
print("Hola", nombre, nombre[5], type(nombre))

nombre = 25
print("Tu edad es: ", type(nombre))

Hola Luis Hernandez H <class 'str'>
Tu edad es:  <class 'int'>


In [ ]:
programacion = True
diseno = False
print("Te gusta Progra", programacion, "Te gusta diseño", diseno, type(programacion))

Te gusta Progra True Te gusta diseño False <class 'bool'>


In [ ]:
edad = int(input("Dime tu edad: "))
if edad>=18: #comparacion da como resultado un valor booleano
  print("Eres un adulto, tienes responsabilidades...")
else:
  print("Eres libre")

Dime tu edad: 25
Eres un adulto, tienes responsabilidades...


In [ ]:
def sumaDosNumeros(num1, num2):
  resp = num1 + num2
  return resp
print( "La suma: ", sumaDosNumeros(2,5) )    

La suma:  7


In [ ]:
#funciones anonimas... 
sumaNumeros = lambda num1, num2: num1 + num2
print( "La suma: ", sumaNumeros(4,6))

La suma:  10


In [ ]:
numeros = [5,4,7,8,-4,-9] #array unidimensional

In [ ]:
def cuadrado(num):
  return num**2

In [ ]:
def cubo(num):
  return num**3

In [ ]:
def pares(num): #num = 2/2 = 1 == 0
  return num%2==0

In [ ]:
def impares(num):
  return num%2!=0

In [ ]:
numCuadrado = list(map(cuadrado, numeros))
numXDos = list( map(lambda num: num*2, numeros) )
print("Elevar al cuadradola lista: ",numCuadrado, numXDos )

Elevar al cuadradola lista:  [25, 16, 49, 64, 16, 81] [10, 8, 14, 16, -8, -18]


In [ ]:
numCubo = list( map(cubo, numeros) )
print(numCubo)

[125, 64, 343, 512, -64, -729]


In [ ]:
#usando filter dejamos solo los pares
pares = list( filter(pares, numeros) )
print(pares)

[4, 8, -4]


In [ ]:
impares = list( filter(impares,numeros) )
print(impares)

[5, 7, -9]


In [ ]:
#conversor de celsius a f
def cf(c):
  return c*9/5+32

In [ ]:
c = float(input("Grados C: "))
print( "G. C a F", cf(c))

Grados C: 27
G. C a F 80.6


In [ ]:
celsius = lambda c: c*9/5 + 32
c = int(input("Grados Celsius A convertir: "))

print( celsius(c) )

Grados Celsius A convertir: 27
80.6


In [ ]:
gradosc = [27,30,35,40,45]
grados = lambda c: c*1.8 + 32
f = list( map(grados, gradosc) )
print(f)

[80.6, 86.0, 95.0, 104.0, 113.0]


**Tarea:** 
*Hacer un algoritmo en python que convierta de farenheit a celsius*
*Hacer un algoritmo en python que convierta de Celsius a Kelvin y viceversa*

In [ ]:
numeros = [5,7,9,6,8,7,3] #, se declara y usa con corchetes

In [ ]:
print(" Lista original ", " Buscando: ", numeros[numeros.index(3)], numeros[0] )

 Lista original   Buscando:  3 5


In [ ]:
numeros[0] = 4
print("Nuevo valor: ", numeros[0])

Nuevo valor:  4


In [ ]:
edades = (2,5,12,18,65,100) #Una tupla es inmutable, se declara con parentesis
print( "Las edades: ", edades, edades[0] )

Las edades:  (2, 5, 12, 18, 65, 100) 2


In [ ]:
#edades[0] = 0 #NO puedo cambiar su valor, y se usa con corchetes
print( "Nuevo Valor", edades[0] )

Nuevo Valor 2


In [ ]:
print("Edades: ", edades, "Filtrar solo mayores de edad: ", tuple(filter( lambda num: num>=18, edades )) )

Edades:  (2, 5, 12, 18, 65, 100) Filtrar solo mayores de edad:  (18, 65, 100)


In [ ]:
buscando = int(input("Valor: "))
print( "Edades: ", edades, "Buscando el numero: ", buscando, "esta en la posicion: ", edades.index(buscando) )

Valor: 5
Edades:  (2, 5, 12, 18, 65, 100) Buscando el numero:  5 esta en la posicion:  1


In [ ]:
notas = [5, 4, 10, 7.8, 9]
print("Mis Notas: ", notas, "Notas sobresalientes: ", list(filter(lambda n:n>=9, notas)) )

Mis Notas:  [5, 4, 10, 7.8, 9] Notas sobresalientes:  [10, 9]


In [ ]:
notasCiclo = [
    [5,6,6],
    [8,7,10],
    [10,10,10]
]
print( "Mis Notas", notasCiclo, "C1: ", notasCiclo[0], "C2: ", notasCiclo[1], "C3: ", notasCiclo[2], "PC1: ", notasCiclo[0][2] )

Mis Notas [[5, 6, 6], [8, 7, 10], [10, 10, 10]] C1:  [5, 6, 6] C2:  [8, 7, 10] C3:  [10, 10, 10] PC1:  6


In [ ]:
materias = [
    [ #Progra
      [5,6,[10,8]],#c1
      [5,7,8],#c2
      [6,7,6] #c3
    ],
    [ #Diseño
      [10,9,10], #c1
      [10,10,8], #c2
      [10,10,10] #c3
    ],
    [ #Redes
      [8,7,8], #c1
      [7,9.23,9], #c2
      [8,7,9]  #c3
    ]
]
print("Progra: C1 - P", (materias[0][0][2][0] + materias[0][0][2][1])/2, "; Redes C2, Lab2: ", materias[2][1][1] )

Progra: C1 - P 9.0 ; Redes C2, Lab2:  9.23


In [ ]:
misMaterias = {} #los diccionarios (json) se definen con llaves y se usan con corchetes
misNotas = {
    "Progra I":{
        "CI": {
            "Lab1": 5,
            "Lab2": 6,
            "Parcial": [10,8]
        },
        "CII":{
            "Lab1": 5,
            "Lab2": 7,
            "Parcial": 8
        },
        "CIII":{
            "Lab1":6,
            "Lab2":7,
            "Parcial":6
        }
    },
    "Diseño":{
        "CI": {
            "Lab1": 10,
            "Lab2": 9,
            "Parcial": 10
        },
        "CII":{
            "Lab1": 10,
            "Lab2": 10,
            "Parcial": 8
        },
        "CIII":{
            "Lab1":10,
            "Lab2":10,
            "Parcial":10
        }
    }
}
print( misNotas["Progra I"]["CI"]["Parcial"], misNotas["Diseño"]["CII"]["Lab2"] )

[10, 8] 10


Conversor de Divisas...
Dolares, Quetzalez, Lempiras, Cordobas, Pesos CR, Euros, Colon SV, Yenes, Soles Peruanos

In [ ]:
divisas = ["Seleccione una divisa", "Dolares", 
           "Quetzalez", "Lempiras", "Cordobas", 
           "Pesos CR", "Euros", "Colon SV", 
           "Yenes", "Soles Peruanos"
]
valores = [0,1, 7.71, 24.51, 35.77, 664.64, 0.98, 8.75, 134.13, 3.84]

In [ ]:
def convertir(de, a, cantidad):
  return valores[a] / valores[de] * cantidad

In [ ]:
print("*** MENU ***")
print("SELECCIONE UNA DIVISA")
print("1. Dolares")
print("2. Quetzalez")
print("3. Lempiras")
print("4. Cordobas")
print("5. Pesos CR")
print("6. Euros")
print("7. Colon SV")
print("8. Yenes")
print("9. Soles Peruanos")

*** MENU ***
SELECCIONE UNA DIVISA
1. Dolares
2. Quetzalez
3. Lempiras
4. Cordobas
5. Pesos CR
6. Euros
7. Colon SV
8. Yenes
9. Soles Peruanos


In [ ]:
de = int(input("DE: "))
a = int(input("A: "))
cantidad = float(input("Cantidad: "))

print( divisas[de],  " a ", divisas[a], " es: ", convertir(de, a, cantidad) )

DE: 6
A: 2
Cantidad: 10
Euros  a  Quetzalez  es:  78.6734693877551


**ESTRUCTURAS DE CONTROL**

In [ ]:
#if dondicional si

edad = int(input("Dime tu edad: "))
if(edad<=2):
  print("Eres un bebe")
elif(edad<12):
  print("Edad un niño")
elif (edad<18):
  print("Eres un adolescente")
elif(edad<65):
  print("Eres mayor de edad, tienes responsabilidades")
elif(edad<80):
  print("Eres un adulto mayor")
else:
  print("Larga vida")

Dime tu edad: 1
Eres un bebe


In [ ]:
#Ciclos, while y for
#obtenga la tabla de multiplicar de x numero dada por el usuario
#while
ntabla = int(input("Tabla: "))
i=1
while(i<=10):
  print(ntabla, "X", i, "=", ntabla*i)
  i+=1


Tabla: 13
13 X 1 = 13
13 X 2 = 26
13 X 3 = 39
13 X 4 = 52
13 X 5 = 65
13 X 6 = 78
13 X 7 = 91
13 X 8 = 104
13 X 9 = 117
13 X 10 = 130


In [ ]:
ntabla = int(input("Tabla: "))
for i in range(1, 11):
  print(ntabla, "X", i, "=", ntabla*i)

Tabla: 13
13 X 1 = 13
13 X 2 = 26
13 X 3 = 39
13 X 4 = 52
13 X 5 = 65
13 X 6 = 78
13 X 7 = 91
13 X 8 = 104
13 X 9 = 117
13 X 10 = 130


In [ ]:
#HAcer un programa en python que genera la tabla de multiplicar dada por el usuario, hasta que el usuario desida salir
s="s"
while s=="s":
  ntabla = int(input("Tabla: "))
  for i in range(1,11):
    print(ntabla, "X", i, "=", ntabla*i)
  s = input("Continuar s, cualquier otra tecla para salir ")
print("Salistes...")

Tabla: 5
5 X 1 = 5
5 X 2 = 10
5 X 3 = 15
5 X 4 = 20
5 X 5 = 25
5 X 6 = 30
5 X 7 = 35
5 X 8 = 40
5 X 9 = 45
5 X 10 = 50
Continuar s, cualquier otra tecla para salir 
Salistes...


TAREA DE LA CEMANA 4

EGERCICIO #1
.Calcular sobre el sueldo las deducciones por ley ISSS (3%), AFP( 6.25%), ISR(Según tabla) e imprima el total deducciones y sueldo neto.


In [ ]:
sueldo = float(input("Ingrese su sueldo: "))

descuentos = [3,6.25,27]



isss = (sueldo*descuentos[0])/100

afp = (sueldo*descuentos[1])/100

isr = (sueldo*descuentos[2])/100

totaldeduciones = isss+afp+isr

total = sueldo-totaldeduciones

print("**DEDUCCIONES**")

print("ISSS: ",isss)

print("AFP: ",afp)

print("ISR: ",isr)

print("**SUELDO TOTAL**")

print("Sueldo Neto: ", total)

EJESISIO #2 Calcular el valor a pagar según el número de metros consumidos de agua potable (Ver tabla en la siguiente diapositiva).



In [ ]:
consumo = int(input("Dijite el valor de consumo del agua:  "))

tarfaFija = 6



if consumo <= 18:

  print("Su Tarifa es: $",tarfaFija)

elif consumo >= 19:

  total = (consumo - 18)*0.45+tarfaFija

  print("Su tarifa es: $",total)

elif consumo >= 29:

  total2 = (consumo - 28)*0.65+tarfaFija

  print("Su tarifa es: $",total2)

else:

  print("Dijitos no validos")

EJERCICIO # 3
Desarrolle la tabla de meses acumulados a n meses. Si puede encontrar la formula matemática para evitar ciclos. 


In [ ]:
def Acumulacion(mes):
  n = mes
  acumulado = n * (n+1) /2
  resultado = acumulado
  print("**MES** ",n, "\n**ACUMULADO** ", resultado)
 
mes = int(input("Ingrese los meses a pagar: "))
 
Acumulacion(mes)

EJERCICIO #4 Conversores: (40%) (10 ítems como mínimo por cada uno)
• Monedas (Resuelto en clases)
• Longitud
• Masa
• Almacenamiento
• Tiempo
• Volumen
• Área

In [ ]:
def covertir(conversor,de,a,cantidad):
  return valores[conversor][a]/valores[conversor][de]*cantidad

In [ ]:
etiquetas = [
[],
  ["Seleccione una divisa", "1.dolares","2.quetazles","3.lempiras","4.cordobas","5.pesos cr",
    "6.euros","7.colon sv","8.yenes","9.soles peruanos","10.pesos colombianos"],#Divisas
 
  ["Seleccione una opcion", "1.Metros", "2.Milimetros", "3.Centiemtros", "4.pie",
    "5.Varas", "6.Yardas", "7.Km", "8.Millas","9.Decimentro","10.Nanometro"],#Longiudes
 
  ["Seleccione la opcion", "1.Gigabyte", "2.Megabyte", "3.Kilobyte","4.Terabyte","5.Petabyte",
    "6.Tebibyte","7.Pebibyte","8.Gibibyte","9.Mebibyte","10.Byte"],#Almacenamiento
 
  ["Seleccione la opcion", "1.Siglo","2.Decada","3.AñoNatural","4.Mes","5.Semana",
    "6.Dia","7.Hora","8.Minuto","9.Segundo","10.Milisegundo"],#Tiempo
 
  ["Seleccione la opcion", "1.Tonelada","2.Kilogramo","3.Gramo","4.Centrigramo","5.Miligramo",
    "6.Microgramo","7.Tonelada Corta","8.Stone","9.Libra","10.Onza"],#Masa
 
  ["Seleccione la opcion", "1.Metro Cubico", "2.Litro","3.Mililitro","4.Galon Imperial","5.Cuarto Imperial",
    "6.Pinta Imperial","7.Taza Imperial","8.Onza Liquida Imperial","9.Cucharada Imperial","10.Cucharadita Imperial"],#Volumen
 
  ["Seleccione la opcion", "1.Kilimetro Cuadrado","2.Metro Cuadrado","3.Milla Cuadrada","4.Yarda Cuadrada","5.Pie Cuadrado",
    "6.Pulgada Cuadrada","7.Hectarea","8.Acre","9.Tareas","10.Manzanas"]#Area,Manzanas,Tareas
]
 
valores = [
    [],
    [0,1,7.71,24.51,35.77,664.64,0.98,8.71,134.13,3.84,4249.81],#Divisas
    [0,1,1000,100, 3.28084,1.120,1.0936133,0.001,0.000621371,10,1000000000],#Longitudes
    [0,1,1000, 1000000,1126,1024,9.0949e-13,1.126,1.049e+6,1.074e+9,1.126e+15],#Almacenamiento
    [0,1,10,100,1200,5214,36500,876000,5.256e+7,3.154e+9,3.154e+12],#tiempo
    [0,1,1000,1e+6,1e+8,1e+9,1e+12,1.102,157.5,22005,35270],#Masa
    [0,1,1000,1e+6,220,879.9,1760,3520,35200,56310,168900],#Volumen
    [0,1,1e+6,0.3861,1.1196e+6,1.076e+7,1.55e+9,100,247.1,1590.3307888,141.7]#Area,Manzanas,Tareas
]

In [ ]:
print("**MENU DE CONVERSIONES**")
print("1.Divisas")
print("2.Longitud")
print("3.Almacenamiento")
print("4.Tiempo")
print("5.Masa")
print("6.Volumen")
print("7.Area\n")
 
opcion = int(input("Conversor: "))
print("CONVERSOR ELEGIDO: ", etiquetas[opcion])
de = int(input("DE: "))
a = int(input("A: "))
cantidad = int(input("Cantidad: "))
print("Conversion de ", etiquetas[opcion][de], " a ", etiquetas[opcion][a], " equivale a ", covertir(opcion,de,a,cantidad))

EJERCICIO #5 Realizar un conversor propio, donde espesifique el valor de conversión: (10%)
• 1 caja -> 25 unidad. Viceversa 80 unidades – 3/5 -> cajas/unidades
• 1 saco -> 50 libras. Viceversa 130 libras – 2/30 -> sacos/libras
• 1 fardo -> 5 unidades. Viceversa 28 unidades – 5/3 -> fardos/unidades

In [ ]:
def convertir(opcion,de, a, cantidad):

  return valor[opcion][a] / valor[opcion][de] * cantidad

In [ ]:
conversor_p = [

  [],

  ["seleccione uno a convertir", "1. Unidades","2. cajas"],

  ["Selecione uno a convertir""3. Saco","4. Libra"],

  ["Seleccione uno a convertir" "5. Fardo", "2. Unidades"]

]

valor = [

  [],

  [0,1,25],

  [0,1,50],

  [0,1,5]

]

In [ ]:
#polimosfismo #pulpo

num1 = 5.5
num2 = 4
print("Suma: ", num1 + num2)

nombre = "Luis "
saludo = "Hola " + nombre
print(saludo)

Suma:  9.5
Hola Luis 


In [ ]:
!pip install pandas

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [2]:
#USO del paradigma del Programacion Orientada a Objetos
class deducciones:
  def __init__(self, sueldo, porcentajeIsss, porcentajeAfp, tablaIsr):
    self.sueldo = sueldo
    self.porcentajeIsss = porcentajeIsss
    self.porcentajeAfp = porcentajeAfp
    self.tablaIsr = tablaIsr
  
  def calcularIsss(self):
    return self.sueldo*self.porcentajeIsss/100

  def calcularAfp(self):
    return self.sueldo*self.porcentajeAfp/100
  
  def calcularIsr(self):
    self.sueldoNeto = self.sueldo - (self.calcularIsss() + self.calcularAfp())
    self.isr = 0
    for tramo in self.tablaIsr:
      if self.sueldoNeto>=tramo[0] and self.sueldoNeto<=tramo[1]:
        self.isr = (self.sueldoNeto - tramo[0])*tramo[2]/100 + tramo[3]
    return self.isr

In [3]:
#ISSS
porcentajeIsssP = 3 #esta valor puede venir de una base de datos
porcentajeIsssE = 7.5
#aFP
porcentajeAfpP = 6.25
porcentajeAfpE = 7.75
#ISR
tablaIsr = [
    [0.01,    472,     0,  0],     #tramo I
    [472.01,  895.24,  10, 17.67], #tramo II
    [895.25,  2038.10, 20, 60],    #tramo III
    [2038.11, 9999999, 30, 288.57] #tramo IV
]

In [6]:
sueldo = float(input("Sueldo: "))

objDp = deducciones(sueldo, porcentajeIsssP, porcentajeAfpP, tablaIsr)
objDe = deducciones(sueldo, porcentajeIsssE, porcentajeAfpE, tablaIsr)

print("ISSS - Personas: ", round(objDp.calcularIsss(),2), "ISSS - Empresa:", round(objDe.calcularIsss(),2))
print("AFP - Personas: ", round(objDp.calcularAfp(),2), "AFP - Empresa: ", round(objDe.calcularAfp(), 2))
print("Isr - Personas: ", round(objDp.calcularIsr(),2) )

Sueldo: 800
ISSS - Personas:  24.0 ISSS - Empresa: 60.0
AFP - Personas:  50.0 AFP - Empresa:  62.0
Isr - Personas:  43.07
